spec/p2p/p2p-top-level.md

```diff
@@ -0,0 +1,310 @@
+1. Connectivity. The overlay network induced by the local neighborhoods (defined by the lists of peers) is sufficiently connected so that the reactors can implement communication on top of it that is sufficient for their needs
+    > There is the design decision that the same overlay is used by all reactors. It seems that consensus has the strongest requirements regarding connectivity and this defines the required properties
+2. Stability. Typically, connections between correct peers should be stable
+    > Even if at every time *t* we satisfy Point 1, if the overlays at times *t* and *t+1* are totally different, it might be hard to implement decent communication on top of it. E.g., Consensus gossip requires a neighbor to know its neighbors *k* state so that it can send the message to *k* that help *k* to advance. If *k* is connected only one second per hour, this is not feasible.
+3. Openness. It is always the case that new nodes can be added to the system
+    > Assuming 1. and 2. holds, this means, there must always be nodes that are willing to add connections to new peers.
+
+
+
+## Sequential Problem statement
+
+> should be English and precise. will be accompanied with a TLA spec.
+
+TODO: This seems to be a research question. Perhaps we can find some simple properties by looking at the peer-to-peer systems academic literature from several years ago?
+
+# Part III - Distributed System
+
+> Introduce distributed aspects 
+
+> Timing and correctness assumptions. Possibly with justification that the
+assumptions make sense, e.g., it is in the interest of a full node to behave
+correctly 
+
+> should have clear formalization in temporal logic.
+
+## Incentives
+
+TODO: 
+- who will follow the protocol who won't
+- validators hiding behind sentries (they have an incentive to not run it)
+- what can be incentives/strategies of bad nodes 
+     - DNS
+     - filling up all your connections and then disconnecting you
+     - feeding your reactors with garbage
+
+general question (is it likely? do we care)
+
+## Computational Model
+
+TODO: partially synchronous systems?
+
+## Distributed Problem Statement
+
+
+### Design choices
+
+> input/output variables used to define the temporal properties. Most likely they come from an ADR
+
+The p2p layer is
+    - running the peer exchange protocol PEX (in a reactor)
+    - using input from the operator (addresses)
+    - responding to other peers wishing to connect
+> the latter might just be the result of the first two points on the other peer
+
+
+TODO: The following two points seem to be implementation details/legacy design decisions
+- communicate to the reactors over the reactor API
+- I/O
+   - dispatch messages incoming from the network to the reactors
+   - send messages incoming from the reactors to the network (the peers the messages should go to) 
+
+
+### Temporal Properties
+
+> safety specifications / invariants in English 
+
+TODO: In a good period, *p* should stay connected with *q*.
```

Comment:
is this really required? What if there are better connections?
I thought about "In a good period, connections eventually stabilize", but even in good periods there may be changes in the network that the p2p layer might want to take into account. And even if not, if we need to ensure openness and a bounded number of connections, good connections may need to be dropped.